packages/fiori/src/Timeline.ts

```diff
@@ -105,51 +117,103 @@ class Timeline extends UI5Element {
 	}
 
 	_onfocusin(e: FocusEvent) {
-		const target = e.target as TimelineItem;
+		let target = e.target as ITimelineItem | ToggleButton;
+
+		if ((target as ITimelineItem).isGroupItem) {
+			target = target.shadowRoot!.querySelector<ToggleButton>("ui5-toggle-button")!;
+		}
 
 		this._itemNavigation.setCurrentItem(target);
 	}
 
 	onBeforeRendering() {
 		this._itemNavigation._navigationMode = this.layout === TimelineLayout.Horizontal ? NavigationMode.Horizontal : NavigationMode.Vertical;
 
+		if (this.items) {
+			for (let i = 0; i < this.items.length; i++) {
+				this.items[i].layout = this.layout;
+				if (this.items[i + 1] && !!this.items[i + 1].icon) {
+					this.items[i].forcedLineWidth = SHORT_LINE_WIDTH;
+				} else if (this.items[i].icon && this.items[i + 1] && !this.items[i + 1].icon) {
+					this.items[i].forcedLineWidth = LARGE_LINE_WIDTH;
+				}
+			}
+		}
+		this._setLastItem();
+		this._setIsNextItemGroup();
+		this.items[0]._firstItemInTimeline = true;
+	}
+
+	_setLastItem() {
+		const items = this.items;
```

Comment:
Perhaps it would be good to have a simple early return :
if (!this.items || !this.items.length) { 
return;
}
this.items[this.items.length - 1]._lastItem = true